sdk/core/azure-core/tests/perf_tests/_test_base.py

```diff
@@ -0,0 +1,354 @@
+
+        self.error_map = {
+            401: ClientAuthenticationError,
+            404: ResourceNotFoundError,
+            409: ResourceExistsError,
+            304: ResourceNotModifiedError,
+        }
+    
+    def _build_sync_pipeline_client(self, auth_policy):
+        sync_core_policies = {
+            "UserAgentPolicy": UserAgentPolicy,
+            "HeadersPolicy": HeadersPolicy,
+            "ProxyPolicy": ProxyPolicy,
+            "NetworkTraceLoggingPolicy": NetworkTraceLoggingPolicy,
+            "HttpLoggingPolicy": HttpLoggingPolicy,
+            "RetryPolicy": RetryPolicy, 
+            "CustomHookPolicy": CustomHookPolicy,
+            "RedirectPolicy": RedirectPolicy,
+            "ContentDecodePolicy": ContentDecodePolicy,
+            "DistributedTracingPolicy": DistributedTracingPolicy,
+            "RequestIdPolicy": RequestIdPolicy,
+            "SensitiveHeaderCleanupPolicy": SensitiveHeaderCleanupPolicy,
+        } # last 4 are not included in default generated policies
+        default_policies = [
+            UserAgentPolicy,
+            HeadersPolicy,
+            ProxyPolicy,
+            NetworkTraceLoggingPolicy,
+            HttpLoggingPolicy,
+            RetryPolicy, 
+            CustomHookPolicy,
+            RedirectPolicy
+        ]
+
+        if self.args.policies is None:
+            # if None, only auth policy is passed in
+            sync_pipeline = Pipeline(
+                transport=self.sync_transport(),
+                policies=[auth_policy]
+            )
+        elif self.args.policies == "all":
+            # if all, autorest default policies + auth policy
+            sync_policies = [policy(sdk_moniker=self.sdk_moniker) for policy in default_policies]
+            sync_policies.append(auth_policy)
+            sync_pipeline = Pipeline(
+                transport=self.sync_transport(),
+                policies=sync_policies
+            )
+        else:
+            # if custom list of policies, pass in custom list + auth policy
+            try:
+                custom_policies = ast.literal_eval(self.args.policies)
+            except ValueError as exc:
+                raise ValueError(f"""Invalid policies list: {self.args.policies}\n."""
+                                 f"""Must be string list of string policies from the following list:\n"""
+                                 f"""{list(sync_core_policies.keys())}""") from exc
+            try:
+                sync_policies = []
+                for policy in custom_policies:
+                    # if ContentDecodePolicy passed in, check for kwargs for that policy
+                    if policy == "ContentDecodePolicy" and self.args.content_decode_policy_kwargs:
+                        policy_kwargs = ast.literal_eval(self.args.content_decode_policy_kwargs)
+                        print(policy_kwargs)
```

Comment:
Probably don't need this

Reply:
sorry, print slipped through - removing!